interface/src/lib.rs

```diff
@@ -654,6 +659,19 @@ impl<'a> HostInterface for MockHost<'a> {
       static_context.nonce,
     )
   }
+
+  fn deploy(&mut self, code: Vec<u8>) -> Address {
+    // template_address := HASH(template_code)
+    let hash = blake3::hash(&code);
+    let hash_bytes = hash.as_bytes().as_slice();
+    let address = Address::try_from(&hash_bytes[..24]).unwrap();
+
+    if self.templates.contains_key(&address) {
+      panic!("template already exists");
```

Comment:
Panic is fine for now, but let's discuss the right protocol for this. It could:

1. succeed, and just be idempotent (but does it charge gas?)
2. result in a failed tx
3. panic
4. something else?

Reply:
Error handling across the VM worries me too, and I think this is a much bigger problem than visible here. There are at least 2 kinds of issues, which are both "handled" by panicking currently:
1. failures in the host callbacks (like in this example). There is currently no way for the host to return an error so the only option is to panic.
2. failures in the VM like here, when a program tries to read from IO more than there is: https://github.com/athenavm/athena/blob/7a5b832084c1c42ecd7a72d7631ab803bf8f1b50/core/src/syscall/hint.rs#L22-L30

Both make it possible to crash the VM and the host process with a crafted guest program. I think we should invest some time in thinking about error handling and use panics and unwraps really carefully.

Reply:
Now that syscalls are fallible (#110), I changed the `HostInterface::deploy` to return an error.